controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,14 +134,335 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+			"imageFeatures":             "layering",
+			"csi.storage.k8s.io/fstype": "ext4",
+			"imageFormat":               "2",
+			"csi.storage.k8s.io/provisioner-secret-name":            "rook-csi-rbd-provisioner",
+			"csi.storage.k8s.io/provisioner-secret-namespace":       r.storageCluster.Namespace,
+			"csi.storage.k8s.io/node-stage-secret-name":             "rook-csi-rbd-node",
+			"csi.storage.k8s.io/node-stage-secret-namespace":        r.storageCluster.Namespace,
+			"csi.storage.k8s.io/controller-expand-secret-name":      "rook-csi-rbd-provisioner",
+			"csi.storage.k8s.io/controller-expand-secret-namespace": r.storageCluster.Namespace,
+		},
+	}
+	storageClassConfiguration = StorageClassConfiguration{
+		storageClass:      storageClass,
+		reconcileStrategy: storagecluster.ReconcileStrategy(managementSpec.ReconcileStrategy),
+		disable:           managementSpec.DisableStorageClass,
+		isClusterExternal: r.storageCluster.Spec.ExternalStorage.Enable,
+	}
+	return storageClassConfiguration
+}
+
+func (r *StorageClassClaimReconciler) newCephFilesystemStorageClassConfiguration(storageClassConfiguration StorageClassConfiguration) StorageClassConfiguration {
+	persistentVolumeReclaimDelete := corev1.PersistentVolumeReclaimDelete
+	allowVolumeExpansion := true
+	managementSpec := r.storageCluster.Spec.ManagedResources.CephFilesystems
+	storageClass := &storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-cephfs", r.storageCluster.Name),
+			Annotations: map[string]string{
+				"description": "Provides RWO and RWX Filesystem volumes",
+			},
+		},
+		Provisioner:   fmt.Sprintf("%s.cephfs.csi.ceph.com", r.storageCluster.Namespace),
+		ReclaimPolicy: &persistentVolumeReclaimDelete,
+
+		// AllowVolumeExpansion is set to true to enable expansion of OCS backed Volumes.
+		AllowVolumeExpansion: &allowVolumeExpansion,
+		Parameters: map[string]string{
+			"clusterID": r.storageCluster.Namespace,
+			"fsName":    fmt.Sprintf("%s-cephfilesystem", r.storageCluster.Name),
+			"csi.storage.k8s.io/provisioner-secret-name":            "rook-csi-cephfs-provisioner",
+			"csi.storage.k8s.io/provisioner-secret-namespace":       r.storageCluster.Namespace,
+			"csi.storage.k8s.io/node-stage-secret-name":             "rook-csi-cephfs-node",
+			"csi.storage.k8s.io/node-stage-secret-namespace":        r.storageCluster.Namespace,
+			"csi.storage.k8s.io/controller-expand-secret-name":      "rook-csi-cephfs-provisioner",
+			"csi.storage.k8s.io/controller-expand-secret-namespace": r.storageCluster.Namespace,
+		},
+	}
+	storageClassConfiguration = StorageClassConfiguration{
+		storageClass:      storageClass,
+		reconcileStrategy: storagecluster.ReconcileStrategy(managementSpec.ReconcileStrategy),
+		disable:           managementSpec.DisableStorageClass,
+		isClusterExternal: r.storageCluster.Spec.ExternalStorage.Enable,
+	}
+	return storageClassConfiguration
+}
```

Comment:
All the same comments as for the `newCephBlockPoolStorageClassConfiguration` func